Foundation/FileManager.swift

```diff
@@ -892,32 +884,43 @@ open class FileManager : NSObject {
         let ti = (TimeInterval(s.st_mtimespec.tv_sec) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtimespec.tv_nsec))
 #elseif os(Android)
         let ti = (TimeInterval(s.st_mtime) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtime_nsec))
+#elseif os(Windows)
+        let ti = (TimeInterval(s.st_mtime) - kCFAbsoluteTimeIntervalSince1970)
 #else
         let ti = (TimeInterval(s.st_mtim.tv_sec) - kCFAbsoluteTimeIntervalSince1970) + (1.0e-9 * TimeInterval(s.st_mtim.tv_nsec))
 #endif
         result[.modificationDate] = Date(timeIntervalSinceReferenceDate: ti)
-        
+
+#if os(Windows)
+        result[.posixPermissions] = NSNumber(value: UInt64(s.st_mode & ~UInt16(ucrt.S_IFMT)))
+#else
         result[.posixPermissions] = NSNumber(value: UInt64(s.st_mode & ~S_IFMT))
+#endif
         result[.referenceCount] = NSNumber(value: UInt64(s.st_nlink))
         result[.systemNumber] = NSNumber(value: UInt64(s.st_dev))
         result[.systemFileNumber] = NSNumber(value: UInt64(s.st_ino))
-        
+
+#if os(Windows)
+        result[.type] = s.st_mode
```

Comment:
This seems wrong as the `s.st_mode` will include the file permissions, can it be unified with the `let type = ...` on line 917?

Reply:
Ah yeah, that was incorrect. It should now properly get the file type.